src/observations/radec_mpc.jl

```diff
@@ -0,0 +1,497 @@
+        # Number regex (columns 1-5)
+        raw"(?P<num>.{5})",
+        # Temporary designation regex (columns 6-12)
+        raw"(?P<tmpdesig>.{7})",
+        # Discovery asterisk regex (column 13)
+        raw"(?P<discovery>.{1})",
+        # Publishable note regex (column 14)
+        raw"(?P<publishnote>.{1})",
+        # Observation technique regex (column 15)
+        raw"(?P<obstech>.{1})",
+        # Year regex + space (columns 16-20)
+        raw"(?P<year>\d{4}) ",
+        # Month regex + space (columns 21-23)
+        raw"(?P<month>\d{2}) ",
+        # Day regex (columns 24-25)
+        raw"(?P<day>\d{2})",
+        # Fraction of days regex (columns 26-32) 
+        raw"(?P<utc>\.[\d\s]{6})",
+        # α hours regex + space (columns 33-35)
+        raw"(?P<α_hrs>\d{2}) ",
+        # α minutes regex + space (columns 36-38)
+        raw"(?P<α_min>\d{2}) ",
+        # α seconds regex (columns 39-44)
+        raw"(?P<α_sec>\d{2}\.[\d\s]{3})",
+        # δ sign regex (column 45)
+        raw"(?P<δ_sgn>\+|\-)",
+        # δ degrees regex + space (columns 46-48)
+        raw"(?P<δ_deg>\d{2}) ",
+        # δ minutes regex + space (columns 49-51)
+        raw"(?P<δ_min>\d{2}) ",
+        # δ seconds regex (columns 52-56)
+        raw"(?P<δ_sec>\d{2}\.[\d\s]{2})",
+        # Info 1 regex (columns 57-65)
+        raw"(?P<info1>.{9})",
+        # Magnitude regex (columns 66-70)
+        raw"(?P<mag>.{5})",
+        # Band regex (column 71)
+        raw"(?P<band>.{1})",
+        # Catalogue regex (column 72)
+        raw"(?P<catalogue>.{1})",
+        # Info 2 regex (columns 73-77)
+        raw"(?P<info2>.{5})",
+        # Observatory code regex (columns 78-80)
+        raw"(?P<obscode>.{3})"
+    ]
+))
+
+@doc raw"""
+    DateTime(year::Int, month::Int, day::Int, utc::T) where {T <: Real}
+
+Construct a `DateTime` type by parts. `utc` is the fraction of day. 
+"""
+function DateTime(year::Int, month::Int, day::Int, utc::T) where {T <: Real}
+    return DateTime(year, month, day) + Microsecond( round(1e6*86_400*utc) )
+end
+
+@doc raw"""
+    ra(hrs::Int, min::Int, sec::T) where {T <: Real}
+    ra(obs::RadecMPC{T}) where {T <: AbstractFloat}
+
+Return the right ascension in rad. 
+"""
+function ra(hrs::Int, min::Int, sec::T) where {T <: Real}
+    # Convert hours minutes seconds to deg
+    α_deg =  15*(hrs + min/60 + sec/3_600)
+    # Convert deg to rad
+    α_rad = deg2rad(α_deg)
+    
+    return α_rad
+
+end
+
+@doc raw"""
+    dec(sgn::String, deg::Int, min::Int, sec::T) where {T <: Real}
+    dec(obs::RadecMPC{T}) where {T <: AbstractFloat}
+
+Return the declination in rad. 
+"""
+function dec(sgn::String, deg::Int, min::Int, sec::T) where {T <: Real}
+    # Convert degrees minutes seconds to degrees
+    if sgn == "+"
+        δ_deg = +(deg + min/60 + sec/3_600)
+    elseif sgn == "-"
+        δ_deg = -(deg + min/60 + sec/3_600)
+    end
+
+    δ_rad = deg2rad(δ_deg)
+
+    return δ_rad
+end
+
+@doc raw"""
+    RadecMPC(m::RegexMatch)
```

Comment:
Sticking to Julia's convention, shouldn't this function be named `radecMPC`?

Reply:
Since the function  is used to create a `RadecMPC` object, I think it is safe to leave it as it is.

Reply:
You are right, it acts as an outer constructor